tests/phpunit/integration/Modules/Sign_In_With_Google/Tag_GuardTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Tag_GuardTest
+ *
+ * @package   Google\Site_Kit\Tests\Modules\Sign_In_With_Google
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+// phpcs:disable PHPCS.PHPUnit.RequireAssertionMessage.MissingAssertionMessage -- Ignoring assertion message rule, messages to be added in #10760
+
+namespace Google\Site_Kit\Tests\Modules\Sign_In_With_Google;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Modules\Sign_In_With_Google\Settings;
+use Google\Site_Kit\Modules\Sign_In_With_Google\Tag_Guard;
+use Google\Site_Kit\Tests\TestCase;
+
+/**
+ * @group Modules
+ * @group Sign_In_With_Google
+ */
+class Tag_GuardTest extends TestCase {
+
+	/**
+	 * Settings object.
+	 *
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Tag_Guard object.
+	 *
+	 * @var Tag_Guard
+	 */
+	private $guard;
+
+	/**
+	 * Site URL before tests.
+	 *
+	 * @var string
+	 */
+	private $reset_site_url;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->settings       = new Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
+		$this->guard          = new Tag_Guard( $this->settings );
+		$this->reset_site_url = get_option( 'siteurl' );
+	}
+
+	public function tear_down() {
+		parent::tear_down();
+		update_option( 'home', $this->reset_site_url );
+		update_option( 'siteurl', $this->reset_site_url );
+		unset( $_SERVER['HTTPS'] );
+		unset( $_SERVER['SCRIPT_NAME'] );
+	}
+
+	public function test_can_activate() {
+		// Set up HTTPS environment and valid client ID.
+		$_SERVER['HTTPS'] = 'on';
+		update_option( 'siteurl', 'https://example.com/' );
+		update_option( 'home', 'https://example.com/' );
+		$this->settings->set( array( 'clientID' => '1234567890.googleusercontent.com' ) );
```

Comment:
Let's use a valid base-subdomain
```suggestion
		$this->settings->set( array( 'clientID' => '1234567890.apps.googleusercontent.com' ) );
```
There's also a 32 char value after the numeric part but we can perhaps keep that abbreviated